src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/View.scala

```diff
@@ -68,10 +68,14 @@ sealed trait View extends Product with Serializable {
     * returning all the other views unchanged.
     * For the case of ''AggregateElasticView of ViewRef[ProjectRef]'', the conversion is successful when the the mapping ''projectRef -> projectLabel'' exists on the ''cache''
     */
-  def labeled[F[_]](implicit cache: DistributedCache[F], F: Monad[F]): EitherT[F, Rejection, View] =
+  def labeled[F[_]](implicit projectCache: ProjectCache[F], F: Monad[F]): EitherT[F, Rejection, View] =
     this match {
-      case AggregateElasticViewRefs(r) =>
-        cache.projectLabels(r.projects).map(projMap => r.copy(value = r.value.map(vr => vr.map(projMap(vr.project)))))
+      case AggregateElasticViewRefs(v) =>
+        val refToLabel = projectCache.getProjectLabels(v.projects)
+        EitherT(refToLabel.map(resultOrFailures(_) match {
+          case Right(res)     => Right(v.copy(value = v.value.map(vr => vr.map(res(vr.project)))))
+          case Left(failures) => Left(LabelsNotFound(failures))
```

Comment:
You can do `resultOrFailures(_).bimap(failures => LabelsNotFound(failures), res => ...)`